Automattic/newspack-plugin
# feat(ia): added emails tab and section to settings page

### All Submissions:


### Changes proposed in this Pull Request:

- Introduces migration of existing Emails Wizard (WP Admin > Newspack > Engagement > Emails) to new location (WP Admin > Newspack > Settings > Emails)
- Fixes bug where dependency (Newspack Newsletters plugin) was not working (Currently this area is completely blank if Newspack Newsletters is not installed).
- Various refactors and modernizations.

### How to test the changes in this Pull Request:

1. Checkout this branch and build assets `git checkout origin/feat/ia-settings-emails && npm run build`
2. Deactivate Newspack Newsletters i.e. `wp plugin deactivate newspack-newsletters` or via Plugins admin page.
3. Navigate to _/wp-admin/admin.php?page=newspack-settings#/emails_
4. Observe you see a prompt to activate Newspack Newsletters plugin:
![Screenshot 2024-07-09 at 09 41 20](https://github.com/Automattic/newspack-plugin/assets/3676975/ba2f6454-e2e2-43d7-9cc5-c17a20799a93)
5. Click "Activate Newspack Newsletters"
6. Once activated page will refresh. Observe you see the following (List will differ depending on how many Newsletters you have stored):
![Screenshot 2024-07-09 at 09 53 17](https://github.com/Automattic/newspack-plugin/assets/3676975/8cf6bfef-7a48-4006-a076-74c894168932)

#### Further testing:
- Confirm "toggling" individual newsletters works as expected.
- Confirm switching between tabs (i.e. "Connections" "Social" etc) and then back to Emails tab works as expected.

### Other information:
